word_segmentation.py

```diff
@@ -707,6 +728,220 @@ def perfom_bayesian_optimization(hunits_lower, hunits_upper, embedding_dim_lower
     print(optimizer.res)
 
 
+def convert_dic_to_json(dic):
```

Comment:
Better if you `import json`: https://docs.python.org/3/library/json.html

Reply:
Use `json` only to write down the dictionary? Or to use it for writing down the whole `weigths.json` file?